File: java/scf-apollo/src/test/java/org/mydotey/scf/apollo/ApolloConfigurationSourceTest.java
package org.mydotey.scf.apollo;

import org.junit.Test;
import org.mydotey.scf.ConfigurationManager;
import org.mydotey.scf.ConfigurationManagerConfig;
import org.mydotey.scf.Property;
import org.mydotey.scf.facade.ConfigurationManagers;
import org.mydotey.scf.facade.StringProperties;

import com.ctrip.framework.apollo.Config;
import com.ctrip.framework.apollo.ConfigService;

/**
 * @author koqizhao
 *
 * Jul 20, 2018
 */
public class ApolloConfigurationSourceTest {

    @Test
    public void testDemo() {
        // create an apollo config
        Config appConfig = ConfigService.getAppConfig();

        // create a scf apollo configuration source
        ApolloConfigurationSourceConfig sourceConfig = new ApolloConfigurationSourceConfig.Builder()
                .setName("apollo-source").setApolloConfig(appConfig).build();
        ApolloConfigurationSource source = new ApolloConfigurationSource(sourceConfig);

        // create scf manager & properties facade tool
        ConfigurationManagerConfig managerConfig = ConfigurationManagers.newConfigBuilder().setName("my-app")
                .addSource(1, source).build();
        ConfigurationManager manager = ConfigurationManagers.newManager(managerConfig);
        StringProperties properties = new StringProperties(manager);

        // use properties
        Property<String, String> appName = properties.getStringProperty("app.name", "unknown");
        System.out.println("app name: " + appName.getValue());

        // add listener for dynamic property
        Property<String, Integer> requestTimeout = properties.getIntProperty("request.timeout", 1000);
        System.out.println("request timeout: " + requestTimeout.getValue());
        requestTimeout.addChangeListener(e -> System.out.println("do something"));
    }

}

File: dotnet/test/MyDotey.SCF.Apollo.Tests/ApolloConfigurationSourceTest.cs
using System;

using Xunit;
using Com.Ctrip.Framework.Apollo;
using MyDotey.SCF.Facade;

namespace MyDotey.SCF.Apollo
{
    /**
     * @author koqizhao
     *
     * Jul 20, 2018
     */
    public class ApolloConfigurationSourceTest
    {
        [Fact]
        public virtual void TestDemo()
        {
            // create an apollo config
            IConfig appConfig = ApolloConfigurationManager.GetAppConfig().Result;

            // create a scf apollo configuration source
            ApolloConfigurationSourceConfig sourceConfig = new ApolloConfigurationSourceConfig.Builder()
                    .SetName("apollo-source").SetApolloConfig(appConfig).Build();
            ApolloConfigurationSource source = new ApolloConfigurationSource(sourceConfig);

            // create scf manager & properties facade tool
            ConfigurationManagerConfig managerConfig = ConfigurationManagers.NewConfigBuilder().SetName("my-app")
                    .AddSource(1, source).Build();
            IConfigurationManager manager = ConfigurationManagers.NewManager(managerConfig);
            StringProperties properties = new StringProperties(manager);

            // use properties
            IProperty<string, string> appName = properties.GetStringProperty("app.name", "unknown");
            Console.WriteLine("app name: " + appName.Value);

            // add listener for dynamic property
            IProperty<string, int?> requestTimeout = properties.GetIntProperty("request.timeout", 1000);
            Console.WriteLine("request timeout: " + requestTimeout.Value);
            requestTimeout.OnChange += (o, e) => Console.WriteLine("do something");
        }
    }
}